api/pkg/transformer/feast/udf_test.go

```diff
@@ -105,3 +106,78 @@ func TestExtractGeohash(t *testing.T) {
 		})
 	}
 }
+
+func TestJsonExtract(t *testing.T) {
+	testJsonString := []byte(`{
+		"details": "{\"merchant_id\": 9001}",
+		"nested": "{\"child_node\": { \"grandchild_node\": \"gen-z\"}}",
+		"not_json": "i_am_not_json_string",
+		"not_string": 1024,
+		"array": "{\"child_node\": { \"array\": [1, 2]}}"
+	}`)
+	var testJsonUnmarshallled interface{}
+	err := json.Unmarshal(testJsonString, &testJsonUnmarshallled)
+	if err != nil {
+		panic(err)
+	}
+
+	tests := []struct {
+		name           string
+		keyJsonPath    string
+		nestedJsonPath string
+		extractedValue interface{}
+		expError       error
+	}{
+		{
+			name:           "should be able to extract value from JSON string",
+			keyJsonPath:    "$.details",
+			nestedJsonPath: "$.merchant_id",
+			extractedValue: float64(9001),
+		},
+		{
+			name:           "should be able to extract value using nested key from JSON string",
+			keyJsonPath:    "$.nested",
+			nestedJsonPath: "$.child_node.grandchild_node",
+			extractedValue: "gen-z",
+		},
+		{
+			name:           "should be able to extract array value using nested key from JSON string",
+			keyJsonPath:    "$.array",
+			nestedJsonPath: "$.child_node.array[0]",
```

Comment:
Can you test wild card access i.e.: `$.child_node.array[*]`

Reply:
Updated!